core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -4164,16 +4166,12 @@ class ReplicaManagerTest {
         mock(classOf[FetchDataInfo])
       }).when(spyRLM).read(any())
 
-      // Get the current type=DelayedRemoteFetchMetrics,name=ExpiresPerSec metric value before fetching
-      val curExpiresPerSec = safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long]
       replicaManager.fetchMessages(params, Seq(tidp0 -> new PartitionData(topicId, fetchOffset, 0, 100000, Optional.of[Integer](leaderEpoch), Optional.of[Integer](leaderEpoch))), UnboundedQuota, fetchCallback)
       // advancing the clock to expire the delayed remote fetch
       timer.advanceClock(2000L)
 
-      // verify the metric value is incremented since the delayed remote fetch is expired
-      TestUtils.waitUntilTrue(() => curExpiresPerSec + 1 == safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long],
-        "The ExpiresPerSec value is not incremented. Current value is: " +
-          safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long])
+      // verify the DelayedRemoteFetchMetrics.expiredRequestMeter.mark is called since the delayed remote fetch is expired
+      verify(mockDelayedRemoteFetchMetricsMeter, times(1)).mark()
```

Comment:
Yes, it indeed fixed the issue. But I found there's a better way to verify it without changing `expiredRequestMeter` from `val` to `var`. We can verify like this:

```
val curExpiresPerSec = DelayedRemoteFetchMetrics.expiredRequestMeter.count()
replicaManager.fetchMessages(params, Seq(tidp0 -> new PartitionData(...)
timer.advanceClock(2000L)
assertTrue(curExpiresPerSec + 1, DelayedRemoteFetchMetrics.expiredRequestMeter.count())
```

Had a test, and it works. Would you please give it a try?
Thanks.

Reply:
thanks @showuon for the suggestion! sure let me try that.